frontend/src/features/admin/access-request/AccessRequestPage.MOTI.test.tsx

```diff
@@ -132,52 +149,64 @@ describe('AccessRequestPage functionality', () => {
       });
     });
 
-    it('displays enabled agencies', () => {
-      const { queryByText } = testRender();
-      expect(queryByText('agencyVal')).toBeVisible();
+    it('does not show success message by default', () => {
+      const component = mount(
+        <Provider store={store}>
+          <Router history={history}>
+            <AccessRequestPage />
+          </Router>
+        </Provider>,
+      );
+      expect(component.find('div.alert').isEmpty).toBeTruthy();
     });
+  });
 
-    it('Does not display disabled agencies', () => {
-      const { queryByText } = testRender();
-      expect(queryByText('disabledAgency')).toBeNull();
-    });
+  it('renders correctly', () => {
+    const { container } = testRender();
+    expect(container.firstChild).toMatchSnapshot();
+  });
 
-    it('displays enabled roles', () => {
-      const { queryByText } = testRender();
-      expect(queryByText('roleVal')).toBeVisible();
-    });
+  it('renders dropdown for roles', () => {
+    const { container } = testRender();
+    const dropdown = container.querySelector(`select[name="role"]`);
+    expect(dropdown).toBeVisible();
+  });
 
-    it('Does not display disabled roles', () => {
-      const { queryByText } = testRender();
-      expect(queryByText('disabledRole')).toBeNull();
-    });
+  it('displays enabled roles', () => {
+    const { queryByText } = testRender();
+    expect(queryByText('roleVal')).toBeVisible();
+  });
 
-    it('Does not display private roles', () => {
-      const { queryByText } = testRender();
-      expect(queryByText('privateRole')).toBeNull();
-    });
+  it('does not display disabled roles', () => {
```

Comment:
is there a test for the agency not being visible? I suppose technically the snapshot would cover it. Might be better to make it explicit. What do you think?

Reply:
Kind of covered by the showAgency functionality, up to you.

Reply:
I'm okay adding a text to explicitly check for non-existence. My thoughts were that the snapshot covers that but you are right that snapshot could be updated and we would loose that check